CONTRIBUTING.md

```diff
@@ -1,48 +1,103 @@
+* The Pull request must have a clear scope outlined in the description, please don't commit changes unrelated to the scope.
+* Refactorings files changed for or related to the Pull request are encouraged. This includes improving code structure and file organisation.
+* The description of the Pull request must also contain information on what functional areas have been touched and should be tested, ideally including a small test manual
+* Appropriate branch name, if possible following standard conventions like git flow (e.g. feature/shiny_new_feature_name).
+* No small Pull requests exclusively fixing single insignificant typos in code comments, one-off formatting mistakes or whitespace. Aggregate Pull requests fixing bigger areas can be accepted.
+* Automated and low effort Pull requests will not be will not be tolerated or accepted (see [Hacktoberfest](https://blog.domenic.me/hacktoberfest/) for an example on why this rule is necessary).
 
-* If you're looking for ideas for a new feature, please check out the upvoted posts in the [Deluge Suggestions area](https://forums.synthstrom.com/categories/deluge-suggestions) on the [Synthstrom Forum](https://forums.synthstrom.com).
+### Code specific
 
-* If you have an idea for a new feature, please check existing [Draft PRs](https://github.com/SynthstromAudible/DelugeFirmware/pulls?q=is%3Apr+is%3Aopen+is%3Adraft) to see if others are working on something similar already.
-    * Even if someone else is doing something similar, there may still be opportunity to contribute. Comment on the draft PR with your questions and suggestions.
+* All project files, especially source files need to have a compatible license with the project. See [LICENSE](LICENSE).
+* There is no written standard on formatting and coding guidelines yet so the current requirement is to make your code look and feel like the rest of the repository and respect the styling decisions taken therein (for example brace position). Clang format checking has been defined in the `.clang-format` file. It is checked in the GitHub Actions to enforce some guidelines. To check for compliance currently Clang 14 should be used since other versions have different behavior.
+* All changes to the firmware have to be tested on a best effort basis to make sure they work as expected and don't break any existing functionality before submitting as ready to merge. This does not apply to Draft Pull requests.
+* All changes need to be compatible with all available hardware variants, this currently includes OLED and 7-Segment.
+* All changes need to be compatible with the currently official toolchain as described in the [Readme](README.md).
+* If a CI system has been established the acceptance of the CI system is also a requirement (comprising probably of compilation and (unit) testing).
```

Comment:
We have build tests now. Those build smoke tests also upload the resulting Artifacts, so we do have per-PR .bins: https://github.com/SynthstromAudible/DelugeFirmware/actions/runs/5354895945

They're configured to only live for 5 days for now though.

Reply:
Incorporated in new commit, thank you for the feedback @bobtwinkles :)